src/backend/joanie/api.py

```diff
@@ -82,3 +82,67 @@ def post(self, request):
         except ObjectDoesNotExist as err:
             return Response(status=404, data={"errors": err.args})
         return Response(serializers.OrderSerializer(order).data)
+
+
+class AddressView(generics.ListAPIView):
+    """
+    API view allows to get all addresses or create or update a new one for a user.
+
+    GET /api/addresses/
+        Return list of all addresses for a user
+
+    POST /api/addresses/ to create or /api/addresses/<address_id>/ to update with expected data:
+        - name: str, address name
+        - address: str
+        - postcode: str
+        - country: str, country code
+        - type: str, billing or shipping (optional)
+        Return new address just created or updated
+
+    DELETE /api/addresses/<address_id>/
+        Delete selected address
+    """
+
+    serializer_class = serializers.AddressSerializer
+    permission_classes = [permissions.IsAuthenticated]
+    instance_field = "address_id"
+
+    def get_queryset(self):
+        """Custom queryset to get user addresses"""
+        user = models.User.objects.get_or_create(username=self.request.user.username)[0]
+        return user.addresses.all()
+
+    def get_instance(self, **kwargs):
+        """Get address instance"""
+        return get_object_or_404(models.Address, pk=kwargs[self.instance_field])
+
+    def do_create(self, request):
+        """Create a new address for user authenticated"""
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        user = models.User.objects.get_or_create(username=self.request.user.username)[0]
+        serializer.save(owner=user)
+        return Response(status=status.HTTP_201_CREATED, data={"data": serializer.data})
+
+    def do_update(self, request, **kwargs):
+        """Update address selected with new data"""
+        obj = self.get_instance(**kwargs)
+        serializer = self.serializer_class(instance=obj, data=request.data)
+        if not serializer.is_valid():
+            return Response(
+                {"errors": serializer.errors}, status=status.HTTP_400_BAD_REQUEST
+            )
+        serializer.save()
+        return Response({"data": serializer.data})
```

Comment:
Should ensure that address is owned by the user which try to update the instance.